Throttle error message on inconsistent encoder timestamps

PeterBowman opened the pull request:
As a follow-up to https://github.com/robotology/yarp/pull/2862, this patch throttles the error message printed whenever an inconsistency in encoder timestamps is detected. The "Encoder timestamps are not consistent! Data will not be published." is logged every few milliseconds and bloats the terminal. I'm using an arbitrary throttle of one second to mitigate this.

@randaz81 I think the nws_ros{.h,.cpp} files at src/devices/ControlBoardWrapper have been accidentally left out after https://github.com/robotology/yarp/pull/2845.  Perhaps you want to delete them first, then I can rebase this patch accordingly. As a side note, the consistency check patch from https://github.com/robotology/yarp/pull/2841 was applied to both nws_yarp and nws_ros in the ControlBoardWrapper directory, but the latter was copied into a different location using a version prior to said changes.

As commented in https://github.com/robotology/yarp/pull/2862, I'm still considering other ways to treat (or perhaps disable) consistency checking due to my particular use case.

PeterBowman commented:
I'm leaving the changes to the ROS device out of this patch since the device itself has been moved to a different repository at https://github.com/robotology/yarp/commit/48a7dd0d493732b296055c37bb8dcb00dc112261.

> As commented in #2862, I'm still considering other ways to treat (or perhaps disable) consistency checking due to my particular use case.

The use case being: some (but not all) wrapped subjoints stop responding and may resume communication with the board at some later point, in which case and in the current state of things the wrapper device is temporarily muted (not publishing state) by the consistency checker. I would like to keep it alive also in this scenario. Note https://github.com/robotology/yarp/pull/2862 allowed unmuting it if **all** subdevices were made available again. Would it be acceptable in a future PR if the board keeped sending state referring to the still working subdevices, while discarding the outdated timestamps in the average value (used as port envelope)?

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/robotology/yarp?pullRequest=2869) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/robotology/yarp?pullRequest=2869) before we can accept your contribution.<br/>**8** out of **9** committers have signed the CLA.<br/><br/>:white_check_mark: randaz81<br/>:white_check_mark: robotology-bot<br/>:white_check_mark: pattacini<br/>:white_check_mark: fbrand-new<br/>:white_check_mark: Nicogene<br/>:white_check_mark: PeterBowman<br/>:white_check_mark: elandini84<br/>:white_check_mark: traversaro<br/>:x: davidegorbani<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/robotology/yarp?pullRequest=2869) it.</sub>

PeterBowman commented:
I have changed the base branch to yarp-3.8 (it looks like it has triggered some unwanted actions, sorry!). I have also added logging component identifiers (`yCIError(name, id(), ...)`) to reflect which wrapper has issued the error among many available. Perhaps those IDs could be extended to more places and other devices as well?
